src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -11,7 +11,7 @@ public class TorpedoStore {
 
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
-
+  private Random generator = new Random();
```

Comment:
This should be static, as now a new Random object generate with every instance. One Random object is enough for all instances.